### PR TITLE
Add ShareDataFrame class

### PR DESCRIPTION
**Pull request by mdonaka**
depends on #239 
# Summary
Add ShareDataFrame class

# Purpose
libclient-py specification changes

# Contents
- Add ShareDataFrame class
- Add new qmpc class

# Testing Methods Performed
- medium test
- CI

### CODE REVIEW

**Review by mdonaka — COMMENTED** [review]
TODO: update test_get_computation_result

**Review by KotaTakahashi9320 — CHANGES_REQUESTED** [review]
@mdonaka 
Thank you！
It's a small detail, but please confirm.

File: packages/client/libclient-py/quickmpc/qmpc.py [hunk 1]
```diff
@@ -35,6 +35,8 @@ class QMPC:
     def __post_init__(self, endpoints: List[str],
                       retry_num: int, retry_wait_time: int):
         logger.info(f"[QuickMPC server IP]={endpoints}")
+        object.__setattr__(self, "_QMPC__qmpc_request", QMPCRequest(
+            endpoints, retry_num, retry_wait_time))
```
[the remark this answers is not in the document]
Comment (mdonaka) [reply packages/client/libclient-py/quickmpc/qmpc.py:39]: > so maybe I don't have to delete it here.
Thank you! OK

**Comment by mdonaka** [comment]
@KotaTakahashi9320 
I fixed. Could you check?

File: scripts/libclient/src/tests_new/test_get_computation_result.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+def test_restore(method: str, args: dict, expected: pd.DataFrame,
+                 sdf):
+    path = "./result"
+    if os.path.isdir(path):
+        shutil.rmtree(path)
+    os.mkdir(path)
+
+    sdf_resulr = getattr(sdf, method)(*args)
+    sdf_resulr.to_csv(path)
+    result = qmpc_new.restore(sdf_resulr.get_id(), path, 3)
```
Comment (KotaTakahashi9320) [inline scripts/libclient/src/tests_new/test_get_computation_result.py:59]: I think it's probably a typo, please correct.
```suggestion
    sdf_result = getattr(sdf, method)(*args)
    sdf_result.to_csv(path)
    result = qmpc_new.restore(sdf_result.get_id(), path, 3)
```

**Review by KotaTakahashi9320 — CHANGES_REQUESTED** [review]
Thanks!
I made one comment.

File: scripts/libclient/src/tests_new/test_get_computation_result.py (hunk 2 above)
Comment (mdonaka) [reply scripts/libclient/src/tests_new/test_get_computation_result.py]: wow! Thank you!

**Review by KotaTakahashi9320 — APPROVED** [review]
LGTM

**Pull request merged**